lib/rspec/mocks/message_expectation.rb

```diff
@@ -367,6 +368,11 @@ module ImplementationDetails
         # @private
         attr_reader :type
 
+        unless defined?(Mutex)
+          Support.require_rspec_support 'mutex'
+          Mutex = Support::Mutex
```

Comment:
Can we refactor this slightly sorry, can we do this:
```Can we refactor this slightly sorry, can we do this:
```
if defined?(Mutex)
  Support::Mutex = Mutex
else
  Support.require_rspec_support 'mutex'
end
```

And change all the calls to `Support::Mutex`

The reasoning being this is a lib file and we don't want to poison the older Rubies. (It would make it seem like `Mutex` is available in code during tests, and then fail in reality).